x/emissions/keeper/msgserver/msg_server_worker_payload.go

```diff
@@ -85,6 +92,13 @@ func (ms msgServer) InsertWorkerPayload(ctx context.Context, msg *types.MsgInser
 	// Append this individual inference to all inferences
 	if msg.WorkerDataBundle.InferenceForecastsBundle.Forecast != nil {
 		forecast := msg.WorkerDataBundle.InferenceForecastsBundle.Forecast
+		isForecasterRegistered, err := ms.k.IsWorkerRegisteredInTopic(ctx, topicId, forecast.Forecaster)
+		if err != nil {
+			return nil, errorsmod.Wrapf(err, "Error address is not registered in this topic")
+		}
+		if !isForecasterRegistered {
+			return nil, errorsmod.Wrapf(err, "Error address is not registered in this topic")
+		}
 		err = ms.k.AppendForecast(ctx, topicId, *nonce, forecast)
```

Comment:
where are we "skimming the top as we go"? I also don't see any spot where we're enforcing a maximum number of bundles across actors per epoch (the actual "skimming" part. maybe I'm missing something?)?

Reply:
The AppendInference method contains this logic afaiu.